woodwork/table_accessor.py

```diff
@@ -38,24 +38,53 @@
 dd = import_or_none('dask.dataframe')
 ks = import_or_none('databricks.koalas')
 
-
+from typing import Optional, Union
+from woodwork.logical_types import LogicalType
 class WoodworkTableAccessor:
     def __init__(self, dataframe):
         self._dataframe_weakref = weakref.ref(dataframe)
         self._schema = None
 
-    def init(self,
-             index=None,
-             time_index=None,
-             logical_types=None,
-             already_sorted=False,
-             schema=None,
-             validate=True,
-             use_standard_tags=True,
-             **kwargs):
-        """Initializes Woodwork typing information for a DataFrame.
+    def init(self, schema=None, **kwargs):
+        """Initializes Woodwork typing information for a DataFrame with only keyword arguments.
+        logical type inference is performed on columns with unspecified logical types"""
+        if schema is not None:
+            raise ValueError("call init_with_partial_schema or init_with_full_schema to pass a schema")
```

Comment:
Didn't really think about this till now, but I wonder we might need to keep `init(schema=schema)` to avoid adding a breaking change. We could call `self.init_with_full_schema(schema=None)` so that it matches the current behavior by just using the schema, assuming it's relevant for the whole table, and ignoring any other parameters.

We'd probably want to add a warning that tells users that they should go through the full or partial schema methods, but I'm thinking that we should avoid the breaking change here if possible.